cpp/velox/utils/ConfigExtractor.cc

```diff
@@ -41,4 +53,103 @@ bool debugModeEnabled(const std::unordered_map<std::string, std::string>& confMa
       std::make_shared<const facebook::velox::core::MemConfigMutable>(confMap);
   return veloxCfg->get<bool>(kDebugModeEnabled, false);
 }
+
+std::shared_ptr<facebook::velox::core::MemConfigMutable> getHiveConfig(
+    const std::shared_ptr<const facebook::velox::Config>& conf) {
+  auto hiveConf = std::make_shared<facebook::velox::core::MemConfigMutable>();
+
+#ifdef ENABLE_S3
+  std::string awsAccessKey = conf->get<std::string>("spark.hadoop.fs.s3a.access.key", "");
+  std::string awsSecretKey = conf->get<std::string>("spark.hadoop.fs.s3a.secret.key", "");
+  std::string awsEndpoint = conf->get<std::string>("spark.hadoop.fs.s3a.endpoint", "");
+  bool sslEnabled = conf->get<bool>("spark.hadoop.fs.s3a.connection.ssl.enabled", false);
+  bool pathStyleAccess = conf->get<bool>("spark.hadoop.fs.s3a.path.style.access", false);
+  bool useInstanceCredentials = conf->get<bool>("spark.hadoop.fs.s3a.use.instance.credentials", false);
+  std::string iamRole = conf->get<std::string>("spark.hadoop.fs.s3a.iam.role", "");
+  std::string iamRoleSessionName = conf->get<std::string>("spark.hadoop.fs.s3a.iam.role.session.name", "");
+
+  std::string awsSdkLogLevel = conf->get<std::string>(kVeloxAwsSdkLogLevel, kVeloxAwsSdkLogLevelDefault);
+
+  const char* envAwsAccessKey = std::getenv("AWS_ACCESS_KEY_ID");
+  if (envAwsAccessKey != nullptr) {
+    awsAccessKey = std::string(envAwsAccessKey);
+  }
+  const char* envAwsSecretKey = std::getenv("AWS_SECRET_ACCESS_KEY");
+  if (envAwsSecretKey != nullptr) {
+    awsSecretKey = std::string(envAwsSecretKey);
+  }
+  const char* envAwsEndpoint = std::getenv("AWS_ENDPOINT");
+  if (envAwsEndpoint != nullptr) {
+    awsEndpoint = std::string(envAwsEndpoint);
+  }
+
+  if (useInstanceCredentials) {
+    hiveConf->setValue(facebook::velox::connector::hive::HiveConfig::kS3UseInstanceCredentials, "true");
```

Comment:
All keys of hive configuration is replaced as `HiveConfig::xxx`. original implementation is magic string value